serializations.md

```diff
@@ -1013,3 +1013,540 @@ NOTE:
+| 0x5d ; 'GAS'
+| 0x5e ; 'ADDRESS'
+| 0x5f ; 'GASPRICE'
+| 0x60 ; 'LOG0'
+| 0x61 ; 'LOG1'
+| 0x62 ; 'LOG2'
+| 0x63 ; 'LOG3'
+| 0x64 ; 'LOG4'
+| 0x65 ; 'SPEND'
+| 0x66 ; 'ORACLE_REGISTER'
+| 0x67 ; 'ORACLE_QUERY'
+| 0x68 ; 'ORACLE_RESPOND'
+| 0x69 ; 'ORACLE_EXTEND'
+| 0x6a ; 'ORACLE_GET_ANSWER'
+| 0x6b ; 'ORACLE_GET_QUESTION'
+| 0x6c ; 'ORACLE_QUERY_FEE'
+| 0x6d ; 'AENS_RESOLVE'
+| 0x6e ; 'AENS_PRECLAIM'
+| 0x6f ; 'AENS_CLAIM'
+| 0x70 ; 'AENS_UPDATE'
+| 0x71 ; 'AENS_TRANSFER'
+| 0x72 ; 'AENS_REVOKE'
+| 0x73 ; 'BALANCE_OTHER'
+| 0x74 ; 'VERIFY_SIG'
+| 0x75 ; 'VERIFY_SIG_SECP256K1'
+| 0x76 ; 'CONTRACT_TO_ADDRESS'
+| 0x77 ; 'AUTH_TX_HASH'
+| 0x78 ; 'ORACLE_CHECK'
+| 0x79 ; 'ORACLE_CHECK_QUERY'
+| 0x7a ; 'IS_ORACLE'
+| 0x7b ; 'IS_CONTRACT'
+| 0x7c ; 'IS_PAYABLE'
+| 0x7d ; 'CREATOR'
+| 0x7e ; 'ECVERIFY_SECP256K1'
+| 0x7f ; 'ECRECOVER_SECP256K1'
+| 0xfa ; 'DEACTIVATE'
+| 0xfb ; 'ABORT'
+| 0xfc ; 'EXIT'
+| 0xfd ; 'NOP'
+| 0xfe ; 'FUNCTION' only used outside of BBs
+| 0xff ; 'EXTREND' reserved for extending instruction set to two bytes, not used yet.
+
+```
+
+
+### AddressingMode
+
+```
+AddressingMode ::=
+  < >
+| LowAddressingMode
+| HighAddressingMode | LowAddressingMode
+
+LowAddressingMode ::=
+  <<< Mode, Mode, Mode, Mode >>> ; Arg3 Arg2 Arg1 Arg0
+
+HighAddressingMode ::=
+  <<< Mode, Mode, Mode, Mode >>> ; Arg7 Arg6 Arg5 Arg4
+
+Mode ::=
+  <<< 00 >>> ; top of stack (or unused if arity < this pos) Arguments empty
+| <<< 01 >>> ; function argument N, N given by Arguments
+| <<< 10 >>> ; variable N or store -N if N is negative, N given by Arguments
+| <<< 11 >>> ; immediate, value given by arguments
+
+```
+
+
+### Arguments
+
+```
+Arguments ::=
+  < >
+| Argument, Arguments
+
+Argument ::=
+  Integer(N) ; For function argument, variable or store.
+| Data(D)    ; For immediates
+```
+
+
+### Data
+
+```
+Data(D) ::=
+  Boolean
+| Integer
+| String
+| Bytes
+| Bits
+| Address
+| ContractAddress
+| Oracle
+| OracleQuery
+| Channel
+| Tuple
+| List
+| Map
+| StoreMap
+| Variant
+| Type
+```
+
+#### Boolean
+```
+Boolean ::=
+  <<<11111111>>> ; True
+| <<<01111111>>> ; False
+```
+
+#### Integer
+```
+Integer(I) ::=
+  <<< 0,  I, 0 >>>                     ; When abs(I) <  64, I >= 0
+| <<< 1, -I, 0 >>>                     ; When abs(I) <  64, I <  0
+| <<< 01101111 >>>, LargeInt(I - 64)   ; When abs(I) >= 64, I >= 0
+| <<< 11101111 >>>, LargeInt(-I - 64)  ; When abs(I) >= 64, I <  0
+
+LargeInt(I) ::=
+  RLP(Unsigned(I))
+
+;; Encode an unsigned int
+;; where 'shr' is bitwise shift right, and 'band' is bitwise and.
+Unsigned(I) ::=
+  << 0 >> ;; When I = 0
+| Unsigned(I shr 8), (I band 0xff)
+
+```
+
+#### String
+A FATE string is just a byte array. The function size(S) gives the
+number of bytes in the byte array.
+
+```
+String(S) ::=
+  <<< 01011111 >>>                        ; when size(S)  =  0 , S is empty ("")
+| <<< size(S), 01 >>>, S                  ; when size(S) <  64
+| <<< 00000001 >>>, Integer(size(S) - 64) ; when size(S) >= 64
```

Comment:
missing string bytes, should be:
```<<< 00000001 >>>, Integer(size(S) - 64), S```

Reply:
Isn't it much shorter without `S`? 😛 
Good catch!